src/Service/FileCache.php

```diff
@@ -0,0 +1,110 @@
+<?php
+declare(strict_types = 1);
+
+namespace App\Service;
+
+/**
+ * This class manages cache for files being translated
+ */
+class FileCache
+{
+    /**
+     * 5 hours
+     */
+    private const CACHE_DURATION = 5 * 60 * 60;
+
+    /** @var Retriever */
+    private $retriever;
+
+    /** @var string */
+    private $directory;
+
+    /**
+     * @param Retriever $retriever
+     * @param string $directory Cache directory
+     */
+    public function __construct(Retriever $retriever, string $directory = '/tmp')
```

Comment:
The directory should default to a subdir of `sys_get_temp_dir()`, but if we use the Symfony cache component that'd be handled automatically.

Reply:
See above.

Reply:
Sure. Let's use `sys_get_temp_dir()` then, or maybe better would be a subdirectory in `var/` in the project directory? That's where other cache files are.

Reply:
Made it use `./var` via service config.

Reply:
Sorry, I totally overlooked that.